src/lib/EventListener/RequestListener.php

```diff
@@ -16,7 +15,12 @@
 
 class RequestListener implements EventSubscriberInterface
 {
-    use ContainerAwareTrait;
+    private $groupsBySiteAccess;
+
+    public function __construct($groupsBySiteAccess)
+    {
+        $this->groupsBySiteAccess = $groupsBySiteAccess;
```

Comment:
note that siteaccess can change. If we don't take from container, ideally class should be adapted to expose setter to refresh setting on siteaccess changes, there are some examples around but I'm not the export on this.

Reply:
I'm not sure that is relevant in this context: what gets injected is the list of groups per siteaccess, not the _current_ siteaccess.

The current siteaccess may change dynamically, but the list of siteaccess/siteaccess groups may not, right ?

Reply:
as I know, current SiteAccess can be changed, but here we fetch only list all of available groups by SiteAccess. Can it be changed during master request?

Reply:
you are right, if it is the global map then it is not subject to issues on this.

Reply:
Based on the name, it is the list of groups for each SA.

Reply:
I think it is a global map, not  groups of the current SiteAccess